kaal-domain/src/main/kotlin/cz/eman/kaal/domain/usecases/UseCase.kt

```diff
@@ -17,7 +21,15 @@ abstract class UseCase<out T, in Params> {
      * @param params Set of input parameters
      * @return type [T] of parameter. In the most common way the [T] is wrapped to a special use-case implementation.
      */
-    suspend operator fun invoke(params: Params): T = doWork(params)
+    suspend operator fun invoke(params: Params): T =
+        try {
+            doWork(params)
+        } catch (ex: CancellationException) {
+            withContext(NonCancellable) {
+                onCancelled(params)
+            }
+            throw ex
+        }
```

Comment:
I think the UC shouldn't contain more than one method (UC == function). This logic can be exist as a function.

Reply:
Replaced with `cancellationAware` function 👌